storage/tests.py

```diff
@@ -1,3 +1,227 @@
-from django.test import TestCase
+import jwt
+import json
+import bcrypt
 
-# Create your tests here.
+from django.test  import Client, TestCase
+from django.utils import timezone
+from django.http import QueryDict, HttpResponse
+
+from my_settings    import SECRET_KEY, ALGORITHM
+from user.models    import User, Character
+from music.models   import *
+from storage.models import Mylist
+
+
+
+class MylistViewTest(TestCase):
+    def setUp(self):
+        password        = '123qwe!@#QWE'
+        hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+        user = User(
+            email         = 'kst6294@gmail.com',
+            password      = hashed_password,
+            date_of_birth = '1991-09-04',
+            phone_number  = '010-1234-1234',
+        )
+
+        user.save()
+        self.access_token = jwt.encode({'id' : User.objects.get(id=user.id).id}, SECRET_KEY, algorithm=ALGORITHM)
+        self.user    = user
+        self.user_id = user.id
+
+        character = Character(
+            name = '순대',
+            user_id = self.user_id,
+        )
+
+        character.save()
+        self.character    = character
+        self.character_id = character.id
+
+        mylist = Mylist(
+            name = timezone.now().strftime('%Y%m%d'),
+            character_id=self.character_id
+        )
+
+        mylist.save()
+
+        self.mylist = mylist
+        self.mylist_id = mylist.id
+        self.updated_date = timezone.now().strftime('%Y%m%d')
+
+
+    def tearDown(self):
+        User.objects.all().delete()
+
+    def test_read_mylist_api_success(self):
+        client = Client()
+
+        headers = {'HTTP_Authorization' : self.access_token, 'HTTP_Character' : self.character_id}
+
+        response = client.get('/storage/mylist', content_type='application/json', **headers)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(),
+                         {
+                             'MESSAGE' : 'SUCCESS',
+                             'RESULTS' : [
+                                 {
+                                     'mylist_id' : self.mylist.id,
+                                     'mylist_name' : self.mylist.name,
+                                     'mylist_updated_date' : self.mylist.updated_at.strftime('%Y-%m-%d'),
+                                 }]})
+
+    def test_create_mylist_success(self):
+        client = Client()
+
+        headers = {'HTTP_Authorization' : self.access_token, 'HTTP_Character' : self.character_id}
```

Comment:
character_id를 headers에 담아서 념겨주는 이유가 있나요?

```suggestion
        headers = {'HTTP_Authorization' : self.access_token, 'HTTP_Character' : self.character_id}
```

Reply:
해당 flo 사이트에서 사용 하는 모든 기능이 user 를 기반으로 하는 것이 아닌 user 에서 파생된 캐릭터로 정해집니다.

예를 들어 캐릭터를 생성하고 보관함에 노래를 넣을 경우, 캐릭터를 변경하면 보관함 리스트도 변경되게 되는데.

이로 인해 character_id 로 토큰을 생성하지 않고, accesstoken 을 받은 후에, 캐릭터 id 가 존재하면 캐릭터 id를 헤더에 담아서 토큰과 같이 보내주려는 이유입니다.

Reply:
이럴 경우에는 데코레이터를 2개로 나눠주세요~

Reply:
2개의 데코레이터 ( Character Authorization, SigninAuthorization ) 으로 나누어 합쳤습니다.